components/core/src/glt/Grep.cpp

```
@@ -393,6 +394,152 @@
+    // First, match the token from front to end.
+    size_t find_start_index{0};
+    bool tokens_contain_variable{false};
+    for (auto const& token : tokens) {
+        auto const& token_str = token.first;
+        bool contains_variable = token.second;
+        size_t found_index = logtype_string.find(token_str, find_start_index);
+        if (string::npos == found_index) {
+            printf("failed to find: [%s] from %s\n",
+                   token_str.c_str(),
+                   logtype_string.substr(find_start_index).c_str());
+            throw;
```

cppcheck

[error] Rethrowing current exception with 'throw;', it seems there is no current exception to rethrow. If there is no current exception this calls std

(rethrowNoCurrentException)